src/controllers/expense.controller.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const expenseService = require('../services/expense.service.js');
+const userService = require('../services/user.service.js');
+
+const getAll = (req, res) => {
+  const {
+    userId: recivedUserId,
+    categories,
+    from,
+    to,
+  } = req.query;
+  const userId = Number(recivedUserId);
+
+  let expenses = expenseService.getAll();
+
+  if (!expenses.length) {
+    res.send([]);
+
+    return;
+  }
+
+  if (userId) {
+    expenses = expenses.filter(expense => expense.userId === userId);
+  }
+
+  if (categories) {
+    const categoriesArray = categories.split(',');
+
+    expenses = expenses.filter(expense => (
+      categoriesArray.includes(expense.category)
+    ));
+  }
+
+  if (from) {
+    const fromDate = new Date(from);
+
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt) >= fromDate
+    ));
+  }
+
+  if (to) {
+    const toDate = new Date(to);
+
+    expenses = expenses.filter(expense => new Date(expense.spentAt) <= toDate);
+  }
```

Comment:
It's a bad idea to keep logic in the controllers

Reply:
Implement that in a separate file.